miio/viomivacuum.py

```diff
@@ -279,113 +344,393 @@ def mop_mode(self) -> ViomiMode:
+
+    @property
+    def hepa_life_left(self) -> int:
+        """HYPA left life percent."""
+        return self.data["hypa_life"]
+
+    @property
+    def charging(self) -> bool:
+        """FIXME: True if device is charging?"""
+        return bool(self.data["is_charge"])
+
+    @property
+    def working(self) -> bool:
+        """FIXME: True if device is working?"""
+        return bool(self.data["is_work"])
+
+    @property
+    def light_state(self) -> bool:
+        """FIXME: True if device ?"""
+        return bool(self.data["light_state"])
+
+    @property
+    def main_brush_hours_left(self) -> timedelta:
+        """Main brush left hours."""
+        return timedelta(self.data["main_brush_hours"])
+
+    @property
+    def main_brush_life_left(self) -> int:
+        """Main brush left life percent."""
+        return self.data["main_brush_life"]
+
+    @property
+    def map_number(self) -> int:
+        """Number of saved maps."""
+        return self.data["map_num"]
+
+    @property
+    def mop_hours_left(self) -> timedelta:
+        """Mop left hours."""
+        return timedelta(self.data["mop_hours"])
+
+    @property
+    def mop_life_left(self) -> int:
+        """Mop left life percent."""
+        return self.data["mop_life"]
+
+    @property
+    def mop_route(self) -> ViomiRoutePattern:
+        """Pattern mode."""
+        return ViomiRoutePattern(self.data["mop_route"])
+
+    @property
+    def order_time(self) -> int:
+        """FIXME: ??? int or bool."""
+        return self.data["order_time"]
+
+    @property
+    def repeat_state(self) -> bool:
+        """Secondary clean up state."""
+        return self.data["repeat_state"]
+
+    @property
+    def side_brush_hours_left(self) -> timedelta:
+        """Side brush left hours."""
+        return timedelta(self.data["side_brush_hours"])
+
+    @property
+    def side_brush_life_left(self) -> int:
+        """Side brush left life percent."""
+        return self.data["side_brush_life"]
+
+    @property
+    def start_time(self) -> int:
+        """FIXME: ??? int or bool."""
+        return self.data["start_time"]
+
+    @property
+    def voice_state(self) -> ViomiVoiceState:
+        """Voice state."""
+        return ViomiVoiceState(self.data["v_state"])
+
+    @property
+    def water_percent(self) -> int:
+        """FIXME: ??? int or bool."""
+        return self.data["water_percent"]
+
+    @property
+    def zone_data(self) -> int:
+        """FIXME: ??? int or bool."""
+        return self.data["zone_data"]
+
 
 class ViomiVacuum(Device):
     """Interface for Viomi vacuums (viomi.vacuum.v7)."""
 
+    def __init__(
+        self,
+        ip: str = None,
+        token: str = None,
+        start_id: int = 0,
+        debug: int = 0,
+        lazy_discover: bool = True,
+        timeout: int = 5,
+    ) -> None:
+        super().__init__(ip, token, start_id, debug, lazy_discover, timeout=0.5)
+        self._cache = {"edge_state": None, "rooms": {}}
```

Comment:
Please avoid overriding the `__init__` if possible. The cache could be initialized during the first use, if it's needed.

Reply:
done

Reply:
In fact I need to set the timeout to 0.5  seconds. Same question as the send method: is there any other way to change the default values ?

Reply:
@rytilahti I think I found a way...
Please look at the change the in device.py file.

Reply:
fixed there: https://github.com/rytilahti/python-miio/pull/884